src/dev/TPS2HB35BQ.cpp

```diff
@@ -0,0 +1,130 @@
+    : en1(en1), en2(en2), latchPin(latch), diagEn(diagEn), diagSelect1(diagSelect1), diagSelect2(diagSelect2), senseOut(adc) {
+    setDiagnostics(DiagMode::OFF);
+}
+
+void TPS2HB35BQ::setPowerSwitchStates(bool powerSwitchOneEnabled, bool powerSwitchTwoEnabled) {
+    if (powerSwitchOneEnabled) {
+        en1.writePin(IO::GPIO::State::HIGH);
+    } else {
+        en1.writePin(IO::GPIO::State::LOW);
+    }
+
+    if (powerSwitchTwoEnabled) {
+        en2.writePin(IO::GPIO::State::HIGH);
+    } else {
+        en2.writePin(IO::GPIO::State::LOW);
+    }
+}
+
+void TPS2HB35BQ::setDiagStateEnabled(bool state) {
+    if (state) {
+        diagEn.writePin(IO::GPIO::State::HIGH);
+    } else {
+        diagEn.writePin(IO::GPIO::State::LOW);
+    }
+}
+
+void TPS2HB35BQ::setLatch(LatchMode mode) {
+    if (mode == LatchMode::LATCHED) {
+        latchPin.writePin(IO::GPIO::State::LOW);
+    } else if (mode == LatchMode::AUTO_RETRY) {
+        latchPin.writePin(IO::GPIO::State::HIGH);
+    }
+}
+
+uint32_t TPS2HB35BQ::readSenseOut() {
+    return senseOut.readRaw();
+}
+
+void TPS2HB35BQ::setDiagnostics(DiagMode diag_mode) {
+    switch (diag_mode) {
+    case OFF:
+        setDiagStateEnabled(false);
+        diagSelect1.writePin(IO::GPIO::State::LOW);
+        diagSelect2.writePin(IO::GPIO::State::LOW);
+        break;
+    case FAULT_STATUS:
+        setDiagStateEnabled(true);
+        diagSelect1.writePin(IO::GPIO::State::HIGH);
+        diagSelect2.writePin(IO::GPIO::State::LOW);
+        break;
+    case CH1CURRENT:
+        setDiagStateEnabled(true);
+        diagSelect1.writePin(IO::GPIO::State::LOW);
+        diagSelect2.writePin(IO::GPIO::State::LOW);
+        break;
+    case CH2CURRENT:
+        setDiagStateEnabled(true);
+        diagSelect1.writePin(IO::GPIO::State::LOW);
+        diagSelect2.writePin(IO::GPIO::State::HIGH);
+        break;
+    case TEMP:
+        setDiagStateEnabled(true);
+        diagSelect1.writePin(IO::GPIO::State::HIGH);
+        diagSelect2.writePin(IO::GPIO::State::LOW);
+        break;
+    }
+}
+
+uint32_t TPS2HB35BQ::getCurrent(uint8_t channelSelect) {
+    if (channelSelect == 1) {
+        setDiagnostics(DiagMode::CH1CURRENT);// Set diagnostic mode to sense current
+    } else {
+        setDiagnostics(DiagMode::CH2CURRENT);// Set diagnostic mode to sense current
+    }
+
+    counts = readSenseOut();
+
+    if (counts >= 4095) {
+        setDiagnostics(DiagMode::OFF);
+    }
+
+    /* volts = (ADC Counts * 3300 kilovolts) / 4096 */
+    uint32_t millivolts = (counts * 3300) / 4096;// Turn ADC counts into milli volts
+    uint32_t milliamps = millivolts / rsns;      // Turn milli volts into milli amps
+
+    /* If current is greater than the current limit latch the sns pin */
+    if (milliamps >= icl) {
+        setDiagnostics(DiagMode::OFF);
+        setLatch(LatchMode::LATCHED);// Latch power switches
+        return -1;
```

Comment:
You are returning and checking for a -1 to indicate an error but your return type is unsigned so you will never get -1. A better value to return to indicate an error could be an integer max value. https://cplusplus.com/reference/cstdint/#macros